src/mlpack/methods/ann/layer/elu.hpp

```diff
@@ -206,27 +210,29 @@ class ELU
    * Computes the first derivative of the activation function.
    *
    * @param x Input data.
+   * @param y Propagated data f(x).
    * @return f'(x)
    */
-  double Deriv(const double y)
+  double Deriv(const double x, const double y)
   {
-    return (y > 0) ? lambda : lambda * (y + alpha);
+    return (x > 0) ? lambda : y + lambda * alpha;
   }
 
   /**
    * Computes the first derivative of the activation function.
    *
-   * @param y Input activations.
-   * @param x The resulting derivatives.
+   * @param x Input data.
+   * @param y Output activations f(x).
+   * @param z The resulting derivatives.
    */
   template<typename InputType, typename OutputType>
   void Deriv(const InputType& x, OutputType& y)
   {
-    y = x;
+    derivative = x;
```

Comment:
Not sure if ``derivative.set_size(arma::size(y))`` is faster as to copy the value, we did go with the copy strategy for some other functions as well, so perhaps it's interesting to take a closer look into this. No need to include this in the PR.